src/connectedk8s/azext_connectedk8s/custom.py

```diff
@@ -489,8 +493,40 @@ def create_connectedk8s(cmd, client, resource_group_name, cluster_name, correlat
                                disable_auto_upgrade, enable_custom_locations, custom_locations_oid,
                                helm_client_location, enable_private_link, arm_metadata,
                                onboarding_timeout, container_log_path)
+    
+    # Add 2nd long running operation to wait for Agent State to reach terminal stage with a default 20 minute timeout window 
+    # TODO: Update poll_for_agent_state method to check for agent state and not provisioning state when feedback loop is implemented
+    if enable_oidc_issuer or enable_workload_identity:
+        print("Waiting for Agent State to reach terminal state")
+        if not poll_for_agent_state(cmd, resource_group_name, cluster_name):
+            print("Timed out waiting for Agent State to reach terminal state")
+            return False
+    
     return put_cc_response
 
+def poll_for_agent_state(cmd, resource_group_name, cluster_name, timeout_minutes=20, interval=5):
```

Comment:
given the long duration, will it be use full to hold  or can we just ask the end user to do a get?

Reply:
I think the whole point of creating the feedback loop is so that in case workload identity enablement fails we also fail cluster onboarding. What is the behavior in AKS if a user enables add-ons? I think they hold till all add ons and components are initialized and in case of a failure cluster onboarding fails.

Reply:
We could shorten the timeout window to 10 minutes if the concern is the cli waiting too long. But I wanted to give enough time for extension installation flow.